web-ui/src/main/resources/catalog/components/admin/uiconfig/ProjectionService.js

```diff
@@ -50,6 +52,17 @@
             }));
       };
 
+      var isDefaultProjection = function(code) {
+        return code === EPSG_LL_WGS84 || code === EPSG_WEB_MERCATOR;
+      }
+
+      var getCachedDef = function(code) {
+        if (!(code in proj4.defs) || isDefaultProjection(code)) {
+          return;
+        }
+        return _cachedDefs[code];
```

Comment:
We don't want to exhaust EPSG.io, so we keep a dictionary of cached Proj4 definitions